contrib/mod_tls.c

```diff
@@ -11929,39 +11929,31 @@ static int tls_netio_close_cb(pr_netio_stream_t *nstrm) {
 
   ssl = (SSL *) pr_table_get(nstrm->notes, TLS_NETIO_NOTE, NULL);
   if (ssl != NULL) {
-    if (nstrm->strm_type == PR_NETIO_STRM_CTRL &&
-        nstrm->strm_mode == PR_NETIO_IO_WR) {
-      if (tls_ctrl_rd_nstrm != NULL) {
-        pr_table_remove(tls_ctrl_rd_nstrm->notes, TLS_NETIO_NOTE, NULL);
+    if (nstrm->strm_type == PR_NETIO_STRM_CTRL) {
+      if (nstrm->strm_mode == PR_NETIO_IO_RD) {
         tls_ctrl_rd_nstrm = NULL;
-      }
 
-      if (tls_ctrl_wr_nstrm != NULL) {
-        pr_table_remove(tls_ctrl_wr_nstrm->notes, TLS_NETIO_NOTE, NULL);
```

Comment:
We don't actually need these `pr_table_remove()` calls here, since the `nstrm->notes` table is automatically cleared, when the `nstrm` pool is destroyed, in the calling `pr_netio_close()` function.

The actual segfault occurred because here, when dealing with the `PR_NETIO_IO_WR` nstrm object, we _also_ try to remove a key from the _read_ `nstrm->notes` table -- and that was already destroyed.  Hence the segfault, due to referencing a stale pointer.